lib/matplotlib/testing/decorators.py

```diff
@@ -298,7 +298,12 @@ def compare(self, idx, baseline, extension):
             remove_ticks_and_titles(fig)
 
         actual_fname = os.path.join(self.result_dir, baseline) + '.' + extension
-        fig.savefig(actual_fname, **self.savefig_kwargs)
+        kwargs = self.savefig_kwargs.copy()
+        if extension == 'pdf':
```

Comment:
No, that test starts a separate subprocess to write the output and doesn't use the image_comparison decorator at all.